### ISSUE TITLE
Slurm nodes with hybrid controller module unable to configure correctly

### ISSUE THREAD

**Issue by saltysoup** [GoogleCloudPlatform/hpc-toolkit#1777]
**If you find a similar existing issue, please comment on that issue instead of creating a new one.**

**If you are submitting a feature request, please start a [discussion](https://github.com/GoogleCloudPlatform/hpc-toolkit/discussions/new?category=ideas-and-feature-requests) instead of creating an issue.**

### Describe the bug

When following the guide for deploying a [hybrid Slurm controller environment](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/docs/hybrid-slurm-cluster/deploy-instructions.md), the nodes being provisioned through the 'cloud' cluster are not brought up due to Slurm install error during bootup.

### Steps to reproduce

1. Follow the deployment guide in above link
2. SSH into the primary controller/login node and provision a new node eg. `salloc -n 1 -p debug`
3. After few min, this generates the following error message:

*On the primary Slurm controller*
```
srun: error: Node failure on gcpcluster-gcpcpu-ghpc-0
srun: error: Nodes gcpcluster-gcpcpu-ghpc-0 are still not ready
srun: error: Something is wrong with the boot of the nodes.
```
*Console output of problematic gcpcluster-gcpcpu-ghpc-0*  
```
*** Slurm is currently being configured in the background. ***

Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: DEBUG: get_metadata: metadata not found (http://metadata.google.internal/computeMetadata/v1/project/attributes/gcpcluster-slurm-devel)
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: DEBUG: fetch_devel_scripts: scripts not found in project metadata, devel mode not enabled
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: INFO: Setting up compute
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: INFO: installing custom scripts: compute.d/ghpc_startup.sh,partition.d/gcpcpu/ghpc_startup.sh
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: DEBUG: install_custom_scripts: compute.d/ghpc_startup.sh
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: DEBUG: install_custom_scripts: partition.d/gcpcpu/ghpc_startup.sh
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: INFO: Set up network storage
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: ERROR: [Errno -2] Name or service not known
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: Traceback (most recent call last):
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/setup.py", line 1247, in <module>
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     main(args)
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/setup.py", line 1223, in main
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     setup(args)
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/setup.py", line 1184, in setup_compute
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     setup_network_storage()
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/setup.py", line 703, in setup_network_storage
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     ext_mounts, int_mounts = partition_mounts(all_mounts)
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/setup.py", line 694, in partition_mounts
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     return separate(internal_mount, mounts)
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/util.py", line 697, in separate
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     return reduce(lambda acc, el: acc[pred(el)].append(el) or acc, coll, ([], []))
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/util.py", line 697, in <lambda>
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     return reduce(lambda acc, el: acc[pred(el)].append(el) or acc, coll, ([], []))
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/setup.py", line 692, in internal_mount
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     return mount_addr == lkp.control_host_addr
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:   File "/slurm/scripts/util.py", line 1174, in control_host_addr
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script:     return socket.gethostbyname(self.cfg.slurm_control_host)
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: socket.gaierror: [Errno -2] Name or service not known
Sep 19 00:02:15 gcpcluster-gcpcpu-ghpc-0 google_metadata_script_runner: startup-script: ERROR: Aborting setup...
```

*deployment blueprint*
https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/docs/hybrid-slurm-cluster/blueprints/hybrid-configuration.yaml

Note that the slurm partition is not using the filestore mount

**Comment by mr0re1** [comment]
Hi @saltysoup , could you please confirm if `static_controller_hostname` is reachable from `compute-vpc-network/primary-subnet`? Can you ping it successfully from VM in this subnet?

**Comment by saltysoup** [comment]
Hi @mr0re1 - Thank you for the suggestion, it looks like the issue was that the startup script I used in slurm partition to add a new entry for the primary controller in /etc/hosts was never executed as slurm was not being installed correctly.

I got it working following your suggestion by creating a new A record with the IP address and FQDN of the primary controller in the VPC's private zone in Cloud DNS. After this, slurm was installed correctly on the new node and everything works as intended.

Closing issue
